include/aws/common/string.h

```diff
@@ -50,6 +52,74 @@ struct aws_string {
 
 AWS_EXTERN_C_BEGIN
 
+#ifdef _WIN32
+/**
+ * For windows only. Converts `to_convert` to a windows whcar format (UTF-16) for use with windows OS interop.
+ *
+ * Note: `to_convert` is assumed to be UTF-8 or ASCII.
+ *
+ * returns NULL on failure.
+ */
+AWS_COMMON_API struct aws_string *aws_string_convert_to_wchar_str(
```

Comment:
Seems like a bad idea to SOMETIMES put UTF-16 in an `aws_string`.

I'd prefer we add a new type like `aws_wstring` or `aws_string16` or `aws_utf16`. We wouldn't need to have the full suite of functions we have for vanilla `aws_string`, just add enough to get by